src/site/apt/matcher_def.apt.vm

```diff
@@ -0,0 +1,364 @@
+    private final String one;
+    
+    @ConfigComponent(type = Component.Type.Parameter, desc = "The second parameter")
+    private final String two;
+    
+    @ConfigComponent(type = Component.Type.Parameter, desc = "The third parameter")
+    private final String three;
+
+    public TPMatcher(String id, String one, String two, String three) { 
+        super(id);
+        if (StringUtils.isEmpthy(one) || StringUtils.isEmpty(two) || StringUtils.isEmpty(three) {
+            throw new ConfigurationException( "None of the three properties (one, two, or three) may be empty");
+        }
+        this.one = one;
+        this.two = two;
+        this.three = three;
+    }
+    
+    public String getOne() { return one; }
+
+    public String getThree() { return two; }
+
+    public String getTwo() { return three; }
+
+    @Override
+    public boolean matches(IHeaders headers) {
+        String text = headers.prune()
+        return text.contains(one) && text.contains(two) && text.contains(three);
+    }
+}
++------------------------------------------+
+
+ The ComponentConfigurations with the parameter type indicate that the members specify properties of the component.  The Matcher must have a "get" method for each parameter and the builder
+ must have a corresponding "set" method.  The names of the methods and the attributes in the XML paarser can be changed by adding a 'name' attribute to the ConfigComponent.
+
+ The builder now looks like:
+
++------------------------------------------+
+package com.example.ratMatcher;
+
+import org.apache.rat.configuration.builders.AbstractBuilder;
+
+public class TPBuilder extends AbstractBuilder {
+
+    private String one;
+    
+    private String two;
+    
+    private String three;
+
+    TPMatcher build() {
+        return new TPMatcher(one, two, three);
+    }
+    
+    public TPBuilder setOne(String one) { this.one = one; }
+
+    public TPBuilder setTwo(String two) { this.two = two; }
+
+    public TPBuilder setThree(String three) { this.three = three; }
+}
++------------------------------------------+
+
+ And the new license file looks like:
+
++------------------------------------------+
+<rat-config>
+    <families>
+        <family id="Joke" name="A joke license" />
+    </families>
+    <licenses>
+        <license family="Joke" id="QSC" name="The QSC license">
+            <qsc/>
+        </license>
+        <license family="Joke" id="TPM" name="The TPM Check">
+            <tpm one="once" two="upon" three="time">
+        </license>
+    </licenses>
+    <matchers>
+        <matcher class="com.example.ratMatcher.QSCBuilder" />
+        <matcher class="com.example.ratMatcher.TPBuilder" />
+    </matchers>
+</rat-config>
++------------------------------------------+
+
+* Embedded matchers.
+
+ It is possible to create matchers that embed other matchers.  The examples in the codebase are the "All", "Any" and "Not" matchers and their associated builders.
```

Comment:
Could we link to the three matchers in Rat (Gitbox):
* All
* Any
* Not

I'd assume that anyone wanting to add new ones would like to take a closer look at these base matchers.